odoo/orm/domains.py

```diff
@@ -897,6 +935,51 @@ def _optimize_field_search_method(self, model: BaseModel) -> Domain:
             model_label=f"{model.env['ir.model']._get(model._name).name!r} ({model._name})",
         ))
 
+    def filter_records(self, records: M) -> M:
+        if not records:
+            return records
+        if self._opt_level < OptimizationLevel.BASIC:
+            return self.optimize(records, full=False).filter_records(records)
+        operator = self.operator
+        if operator not in STANDARD_CONDITION_OPERATORS:
```

Comment:
Make the condition explicit for child_of/parent_of and add a TODO for them, we should implement them in Python at the end (not in this PR).